test/bench/custom/main.mjs

```diff
@@ -20,20 +20,71 @@ const systemInfo = () =>
   ].join('\n');
 console.log(systemInfo());
 
-const suite = new benchmark.Suite();
-
-for (const bench of benchmarks) suite.add(bench.name, bench);
-
-suite
-  .on('cycle', function logBenchmark(event) {
-    console.log(String(event.target));
-  })
-  .on('complete', function outputPerfSend() {
-    const data = Array.from(this).map(bench => ({
-      info: { test_name: bench.name },
-      metrics: [{ name: 'ops_per_sec', value: bench.hz }]
-    }));
-    console.log(util.inspect(data, { depth: Infinity, colors: true }));
-    fs.writeFileSync('customBenchmarkResults.json', JSON.stringify(data), 'utf8');
-  })
-  .run();
+function logBenchmark(event) {
+  console.log(String(event.target));
+}
+
+function processBenchmarkResult(bench, tags, metadata) {
+  return {
+    info: { test_name: bench.name },
+    metrics: [{ name: 'ops_per_sec', value: bench.hz, metadata }],
+    tags
+  };
+}
+
+let completedSuites = 0;
+function completeSuite() {
+  const metadata = { improvement_direction: 'up' };
+  if (++completedSuites >= collectedSuites.length) {
+    let cpuBaselineResults;
+    try {
+      cpuBaselineResults = JSON.parse(fs.readFileSync(`${__dirname}/../etc/cpuBaseline.json`));
+    } catch {
+      throw new Error("Couldn't find baseline results");
+    }
+
+    const cpuBaselineResult = cpuBaselineResults.hz;
+    if (typeof cpuBaselineResult !== 'number') {
+      throw new Error("Couldn't find baseline result");
+    }
+
+    const data = [];
+    for (const { suite, suiteConfig } of collectedSuites) {
+      const { tags } = suiteConfig;
+      for (const bench of Array.from(suite)) {
+        const result = processBenchmarkResult(bench, tags, metadata);
+        result.metrics.push({
+          name: 'normalized_throughput',
+          value: bench.hz / cpuBaselineResult,
+          metadata
+        });
+        data.push(result);
+      }
```

Comment:
Note here that the normalized throughput here is the ops_per_second of the actual benchmark over the ops_per_second of the baseline benchmark